File: src/main/java/micronaut/playground/client/HelloClient.java
package micronaut.playground.client;

import io.micronaut.http.annotation.Get;
import io.micronaut.http.client.annotation.Client;
import io.reactivex.Single;

import javax.validation.constraints.NotEmpty;

@Client("/")
public interface HelloClient {

  @Get("/{name}")
  Single<Hello> hello(@NotEmpty String name);
}

File: src/main/kotlin/micronaut/playground/client/HelloClient.kt
package micronaut.playground.client

import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Get
import io.micronaut.http.client.annotation.Client
import io.micronaut.core.async.annotation.SingleResult
import micronaut.playground.client.model.Hello
import reactor.core.publisher.Mono

@Client("/api/hello")
interface HelloClient {

    @Get("/{name}", consumes = [MediaType.APPLICATION_JSON])
    @SingleResult
    fun hello(name: String): Mono<Hello>
}